languages/kotlin/docs/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,122 @@
-# How to implement an Kotlin concept exercise
+# How to implement a Kotlin concept exercise
 
-TODO: describe how to implement a concept exercise for the Kotlin track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes the steps required to implement a concept exercise in any v3 track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<exercise-slug>`: the name of the exercise in kebab-case (e.g. `functions-basic`).
+- `<ExerciseSlug>`: the name of the exercise in PascalCase (e.g. `FunctionsBasic`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the [exercise's GitHub issue](https://github.com/exercism/v3/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Atrack%2Fkotlin+Implement+new+concept+exercise+in%3Atitle).
+
+Any concept exercise in any v3 track requires the following files to be created:
```

Comment:
There are lots of Kotlin-specifics in the following tree structure, which is not applicable to *any v3 track*. Do we really need to emphasise the intruction overwhelming nature here (and in the first sentence)?

Reply:
Sorry, didn't get your question. I think we'll try to use Kotlin-specific docs wherever we can and refer to generic ones for the rest.

Reply:
My point is there is a plain textual discrepancy between this paragraph and the tree it explains: the tree cannot be applied to *any concept exercise in any v3 track*. The tree is Kotlin-specific.  It I were the reader, I'd prefer direct instruction that do not require deduction. 
It's ok for the first version though.

Reply:
I guess that's totally the point of this docs - to be language-specific (as they are in `languages/kotlin/` directory). :)

Previous paragraph also has link to kotlin-specific issues.

Reply:
And now, if we agreed on that the document should be entirely dedicated to *how to do it in Kotlin track* - shouldn't we stop insisting on *any exercise for any language* and *generic* nature if the document? What are your thoughts?

Reply:
Will DM you in Slack to avoid miscommunication. :)